<a href="https://colab.research.google.com/github/pabloivanmorenorivas-design/cursos-data-analysis/blob/main/curso-kodree/pandas/01_getting_started.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Damos inicio con Python Pandas

In [1]:
# Importamos librería de pandas
import pandas as pd

# Leemos archivo csv
path = "https://github.com/pabloivanmorenorivas-design/cursos-data-analysis/releases/download/v1.0-datasets-kodree/sales_data_updated.csv"
sales = pd.read_csv(path)

In [20]:
# Mostrar las primeras 5 filas
sales.head()

,order_date,time,aging,customer_id,gender,device_type,customer_login_type,product_category,product,sales,quantity,discount,profit,shipping_cost,order_priority,payment_method
0,2018-01-01,10:11:40,5.0,29317,Male,Web,Member,Auto & Accessories,Car Media Players,140.0,4.0,0.3,43.2,4.3,Medium,e_wallet
1,2018-01-01,22:30:44,7.0,42270,Male,Web,Member,Auto & Accessories,Car Pillow & Neck Rest,231.0,5.0,0.1,139.5,13.9,High,money_order
2,2018-01-01,21:55:31,10.0,14563,Male,Web,Member,Auto & Accessories,Car Speakers,211.0,5.0,0.1,120.5,12.0,High,credit_card
3,2018-01-01,13:57:15,9.0,58601,Male,Web,Member,Auto & Accessories,Tyre,250.0,4.0,0.2,150.0,15.0,Critical,credit_card
4,2018-01-01,15:17:41,2.0,48342,Male,Web,Member,Auto & Accessories,Tyre,250.0,1.0,0.1,165.0,16.5,High,credit_card


In [21]:
# Para darnos un mejor vistazo de nuestro DataFrame usaremos un método llamado info
sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   order_date           51290 non-null  object 
 1   time                 51290 non-null  object 
 2   aging                51289 non-null  float64
 3   customer_id          51290 non-null  int64  
 4   gender               51290 non-null  object 
 5   device_type          51290 non-null  object 
 6   customer_login_type  51290 non-null  object 
 7   product_category     51290 non-null  object 
 8   product              51290 non-null  object 
 9   sales                51289 non-null  float64
 10  quantity             51288 non-null  float64
 11  discount             51289 non-null  float64
 12  profit               51290 non-null  float64
 13  shipping_cost        51289 non-null  float64
 14  order_priority       51288 non-null  object 
 15  payment_method       51290 non-null 

In [7]:
# ¿Cuál es el número de columnas y filas?
print(f"Hay un total de {sales.shape[0]} filas y {sales.shape[1]} columnas")

Hay un total de 51290 filas y 16 columnas


In [9]:
# ¿Cuál es el tipo de dato de cada columna?
print(sales.dtypes)

order_date              object
time                    object
aging                  float64
customer_id              int64
gender                  object
device_type             object
customer_login_type     object
product_category        object
product                 object
sales                  float64
quantity               float64
discount               float64
profit                 float64
shipping_cost          float64
order_priority          object
payment_method          object
dtype: object


¿Se reconocen correctamente todas las columnas numéricas?
- No, las columnas de aging y quantity deberían ser integers en lugar de floats
- Respecto a otros tipos de datos, las columnas de order date se podrían juntar y convertirlas en datetime]

In [17]:
# ¿Hay valores faltantes?
total_nulos = sales.isna().sum()
print("Las siguientes columnas presentan valores nulos y su valor indica la cantidad de valores faltantes:")
total_nulos[total_nulos > 0]

Las siguientes columnas presentan valores nulos y su valor indica la cantidad de valores faltantes:


,0
aging,1
sales,1
quantity,2
discount,1
shipping_cost,1
order_priority,2


In [18]:
# Muestra los nombres de todas las columnas
sales.columns

Index(['order_date', 'time', 'aging', 'customer_id', 'gender', 'device_type',
       'customer_login_type', 'product_category', 'product', 'sales',
       'quantity', 'discount', 'profit', 'shipping_cost', 'order_priority',
       'payment_method'],
      dtype='object')

In [19]:
# Usa el método describe() para mostrar estadísticas descriptivas de las columnas numéricas
sales.describe()

,aging,customer_id,sales,quantity,discount,profit,shipping_cost
count,51289.000000,51290.000000,51289.000000,51288.000000,51289.000000,51290.000000,51289.000000
mean,5.255035,58155.758764,152.340872,2.502983,0.303821,70.407226,7.041557
std,2.959948,26032.215826,66.495419,1.511859,0.131027,48.729488,4.871745
min,1.000000,10000.000000,33.000000,1.000000,0.100000,0.500000,0.100000
25%,3.000000,35831.250000,85.000000,1.000000,0.200000,24.900000,2.500000
50%,5.000000,61018.000000,133.000000,2.000000,0.300000,59.900000,6.000000
75%,8.000000,80736.250000,218.000000,4.000000,0.400000,118.400000,11.800000
max,10.500000,99999.000000,250.000000,5.000000,0.500000,167.500000,16.800000


In [29]:
sales["aging"].unique()

array([ 5. ,  7. , 10. ,  9. ,  2. ,  1. ,  3. ,  6. ,  8. ,  4. ,  nan,
       10.5])

# Conclusiones del Análisis Exploratorio (EDA)

### 1. Dimensión del Dataset
El conjunto de datos cuenta con un total de **51,290 filas y 16 columnas**, lo que proporciona un volumen sólido de registros para el análisis de ventas.

### 2. Tipos de Datos y Consistencia
* **Variables numéricas forzadas:** Las columnas se reconocieron en su mayoría de forma automática. Sin embargo, campos como `aging` y `quantity` se identificaron como decimales (`float64`) debido a la presencia de valores nulos (`NaN`). En un entorno de producción, lo ideal sería transformarlas a enteros (`int64`) tras realizar la limpieza.
* **Variables temporales:** Las columnas de fecha (`order_date`) y hora (`time`) se cargaron como objetos de texto (`object`). Para análisis temporales avanzados, se recomienda combinarlas y convertirlas al tipo `datetime`.

### 3. Valores Faltantes
Se detectó una cantidad mínima de valores faltantes distribuidos en las siguientes variables:
* `aging` (1)
* `sales` (1)
* `quantity` (2)
* `discount` (1)
* `shipping_cost` (1)
* `order_priority` (2)

Al representar menos del **0.01%** del total de los datos, no afectan significativamente el sesgo del dataset y pueden ser fácilmente imputados o removidos durante la fase de limpieza.

### 4. Valores Inusuales o Anómalos Detectados
* **Anomalía en `aging`:** Al inspeccionar los valores únicos de la columna mediante `.unique()`, se identificó un registro aislado con el valor de `10.5`, mientras que el resto de los datos son números enteros estrictos del 1 al 10. Esto se diagnostica como un error de captura o ruido didáctico.
* **Anomalía en `shipping_cost`:** El análisis descriptivo reveló un costo de envío mínimo de **0.10 dólares**. Considerando que la media es de `7.04` dólares y el 75% supera los `2.50` dólares, esta tarifa resulta sospechosamente baja y podría representar un error de registro o una promoción específica no estandarizada.
* **Ruido en Identificadores:** El método `.describe()` procesó la columna `customer_id` por ser numérica, generando promedios y desviaciones estándar que carecen de valor analítico al tratarse de una variable categórica de identificación.